zulipterminal/ui_tools/boxes.py

```diff
@@ -1112,26 +1112,39 @@ def reactions_view(self, reactions: List[Dict[str, Any]]) -> Any:
         if not reactions:
             return ""
         try:
+            my_user_id = self.model.user_id
             reaction_stats = defaultdict(set)
             for reaction in reactions:
                 user_id = int(reaction["user"].get("id", -1))
                 if user_id == -1:
                     user_id = int(reaction["user"]["user_id"])
-                reaction_stats[reaction["emoji_name"]].add(user_id)
+                user_name = reaction["user"]["full_name"]
+                if user_id == my_user_id:
+                    user_name = "You"
+                reaction_stats[reaction["emoji_name"]].add((user_id, user_name))
 
             sorted_stats = sorted(
-                (reaction, count) for reaction, count in reaction_stats.items()
+                (reaction, list(count)) for reaction, count in reaction_stats.items()
             )
 
-            my_user_id = self.model.user_id
+            for stats in sorted_stats:
+                stats[1].sort(key=lambda id: id[0])
+                if (my_user_id, "You") in stats[1]:
+                    stats[1].remove((my_user_id, "You"))
+                    stats[1].append((my_user_id, "You"))
+
+            translation = {39: None}
             reaction_texts = [
                 (
-                    "reaction_mine" if my_user_id in ids else "reaction",
-                    f":{reaction}: {len(ids)}",
+                    "reaction_mine"
+                    if my_user_id in [id[0] for id in ids]
+                    else "reaction",
+                    f":{reaction}: {len(ids)}"
+                    if len(ids) > 3
+                    else f":{reaction}: {str([id[1] for id in ids]).strip('[]').translate(translation)}",
```

Comment:
Try using `str.join()` here instead. It'll be much simpler to read and also avoid magic numbers like `39`